### PR TITLE
Fix some exception handling

### PR DESCRIPTION
**Pull request by JJ-Pineda**
Fixes some mishandled ptrs. Specifically, exceptions were not being captured by the `atom` `update_property_cache()` method as well as the `TautomerEnumerator` `canonicalize` method.

### CODE REVIEW

File: .github/workflows/test_suite.yml [hunk 1]
```diff
@@ -39,10 +39,23 @@ jobs:
...
+      - name: Install dependencies
         run: |
           sudo apt-get update
-          sudo apt-get install -y build-essential librdkit-dev libssl-dev libboost1.74-dev libboost-serialization1.74-dev pkg-config
+          sudo apt-get install -y build-essential libssl-dev libboost-iostreams-dev libboost-serialization-dev pkg-config curl tar
+
+      - name: Install RDKit
+        run: |
+          cd /tmp
+          if [ $(dpkg --print-architecture) = "amd64" ]; then
```
Comment (xrl) [inline .github/workflows/test_suite.yml:50]: bash `if` statements are tricky. can you try `if [[ $(...) == "amd64"]]; then`

File: .github/workflows/test_suite.yml [hunk 2]
```diff
@@ -39,10 +39,23 @@ jobs:
...
+            curl -O https://rdkit-rs-debian.s3.eu-central-1.amazonaws.com/rdkit_2024_03_3_ubuntu_14_04_amd64.tar.gz
+            tar xf rdkit_2024_03_3_ubuntu_14_04_amd64.tar.gz
+          else
+            curl -O https://rdkit-rs-debian.s3.eu-central-1.amazonaws.com/rdkit_2024_03_3_ubuntu_14_04_arm64.tar.gz
+            tar xf rdkit_2024_03_3_ubuntu_14_04_arm64.tar.gz
+          fi
+          cp -r /tmp/rdkit-Release_2024_03_3/Code /usr/local/include/rdkit
+          cp /tmp/rdkit-Release_2024_03_3/build/lib/* /usr/lib/
```
Comment (xrl) [inline .github/workflows/test_suite.yml:58]: use `mv` to make sure there's only one version of these lib/headers in the filesystem

Comment (JJ-Pineda) [reply .github/workflows/test_suite.yml]: Done

File: .github/workflows/test_suite.yml (hunk 1 above)
Comment (JJ-Pineda) [reply .github/workflows/test_suite.yml:50]: Weirdly, when I made this change, the build failed. The error complained about the *.so files being incompatible, which makes me thing it started getting the wrong tar ball.

**Review by xrl — APPROVED** [review]

**Pull request merged**